client/rpc/src/eth/execute.rs

```diff
@@ -205,29 +210,104 @@ where
 					Ok(Bytes(info.value))
 				} else if api_version == 4 {
 					// Post-london + access list support
-					let access_list = access_list.unwrap_or_default();
-					let info = api
-						.call(
-							&id,
-							from.unwrap_or_default(),
-							to,
-							data,
-							value.unwrap_or_default(),
-							gas_limit,
-							max_fee_per_gas,
-							max_priority_fee_per_gas,
-							nonce,
-							false,
-							Some(
-								access_list
-									.into_iter()
-									.map(|item| (item.address, item.storage_keys))
-									.collect(),
-							),
-						)
+					let encoded_params = sp_api::Encode::encode(&(
+						&from.unwrap_or_default(),
+						&to,
+						&data,
+						&value.unwrap_or_default(),
+						&gas_limit,
+						&max_fee_per_gas,
+						&max_priority_fee_per_gas,
+						&nonce,
+						&false,
+						&Some(
+							access_list
+								.unwrap_or_default()
+								.into_iter()
+								.map(|item| (item.address, item.storage_keys))
+								.collect::<Vec<(sp_core::H160, Vec<H256>)>>(),
+						),
+					));
+
+					// set custom storage override
+					let mut overlayed_changes = sp_api::OverlayedChanges::default();
+					if let Some(state_override) = state_override {
+						let address = from.unwrap_or_default();
+						if let Some(runtime_state_override) = self.runtime_state_override.as_ref() {
+							let hash =
+								self.client.expect_block_hash_from_id(&id).map_err(|err| {
+									internal_err(format!("failed retrieving block hash: {:?}", err))
+								})?;
+
+							runtime_state_override.set_overlayed_changes(
+								self.client.as_ref(),
+								&mut overlayed_changes,
+								hash,
+								api_version,
+								address,
+								state_override.balance,
+								state_override.nonce,
+							);
+						}
+
+						if let Some(code) = &state_override.code {
+							let mut key = [twox_128(PALLET_EVM), twox_128(EVM_ACCOUNT_CODES)]
+								.concat()
+								.to_vec();
+							key.extend(blake2_128(address.as_bytes()));
+							key.extend(address.as_bytes());
+							overlayed_changes.set_storage(key, Some(code.clone().into_vec()));
+						}
+
+						// Prioritize `state_diff` over `state`
+						if let Some(state_diff) = &state_override.state_diff {
+							for (k, v) in state_diff {
+								overlayed_changes.set_storage(
+									k.as_bytes().to_owned(),
+									Some(v.as_bytes().to_owned()),
+								);
+							}
+						} else if let Some(state) = &state_override.state {
```

Comment:
I don't think we can support overriding the "complete" state with our current approach :(